app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt

```diff
@@ -0,0 +1,198 @@
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.core.text.isDigitsOnly
+import com.github.se.travelpouch.model.Location
+import com.github.se.travelpouch.model.activity.Activity
+import com.github.se.travelpouch.model.activity.ActivityModelView
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.GregorianCalendar
+import java.util.Locale
+
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun AddActivity(activityModelView: ActivityModelView){
+
+    var title by remember { mutableStateOf("") }
+    var description by remember { mutableStateOf("") }
+    var dateText by remember { mutableStateOf("") }
+    var location by remember { mutableStateOf("Placeholder") }
+
+    val context = LocalContext.current
+
+    val placeholerLocation = Location(
+        0.0,
+        0.0,
+        Timestamp(0, 0),
+        "name"
+    )
+
+    Scaffold {
+        Column(
+            modifier = Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(16.dp)
+        ) {
+            OutlinedTextField(
+                value = title,
+                onValueChange = { title = it },
+                label = { Text("Title") },
+                placeholder = { Text("Title") }
+            )
+
+            OutlinedTextField(
+                value = description,
+                onValueChange = { description = it },
+                label = { Text("Description") },
+                placeholder = { Text("Description") }
+            )
+
+            OutlinedTextField(
+                value = dateText,
+                onValueChange = {
+                    if(it.isDigitsOnly() && it.length <= 8){
+                        dateText = it
+                    }
+                },
+                label = { Text("Date") },
+                placeholder = { Text("01/01/1970") },
+                visualTransformation = DateVisualTransformation(),
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number)
+            )
+
+            OutlinedTextField(
+                value = location,
+                onValueChange = {  },
+                label = { Text("Location") },
+                placeholder = { Text("Location") }
+            )
+
+            Button(
+                enabled = location.isNotBlank() && title.isNotBlank() && description.isNotBlank()
+                        && dateText.isNotBlank(),
+                onClick = {
+                    try{
+
+                        val dateFormat =
+                            SimpleDateFormat("dd/MM/yyyy", Locale.getDefault()).apply {
+                                isLenient = false // strict date format
```

Comment:
it would be better to add a helper function for the parsing date

Reply:
Yes, you are right.